### PR TITLE
Implement stable hook configuration

### PR DESCRIPTION
**Pull request by arendjr**
## Summary

User-provided React hooks can now be configured to return stable results.

Implements #1128.

I went for a slightly different configuration syntax than suggested in #1128 to avoid repetition of hooks that have both stable results and a dependencies array to be validated.

## Test Plan

Added and updated test cases.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -79,6 +79,34 @@ our [guidelines for writing a good changelog entry](https://github.com/biomejs/b
 
   Contributed by @arendjr
 
+- Implemented [#1128](https://github.com/biomejs/biome/issues/1128). User-provided React hooks can
+  now be configured to return stable results. For example:
```
Comment (ematipico) [inline CHANGELOG.md:83]:
```suggestion
  now be configured to track stable results. For example:
```

**Review by ematipico — CHANGES_REQUESTED** [review]
The documentation of this rule should be update to document the new options and explain how they work. Looking at the changelog, for example, it's not clear what is array and what are the numbers inside the array.

**Comment by arendjr** [comment]
Docs are updated.

**Review by ematipico — APPROVED** [review]
I wouldn't mind having this in v1.6 :)

**Pull request merged**